components/menu/MenuPopover.tsx

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Instance} from '@popperjs/core';
+import React, {useEffect} from 'react';
+import {usePopper} from 'react-popper';
+
+import RootPortal from 'components/root_portal';
+
+import {MenuPopoverProps} from './Menu.types';
+
+export const useUpdateOnVisibilityChange = (
+    update: Instance['update'] | null,
+    isVisible?: boolean,
+) => {
+    const updateComponent = async () => {
+        if (!update) {
+            return;
+        }
+        await update();
+    };
+
+    useEffect(() => {
+        if (!isVisible) {
+            return;
+        }
+        updateComponent();
+    }, [isVisible]);
+};
+
+const MenuPopover = ({
+    triggerRef,
+    isVisible,
+    isMobile,
+    placement = 'auto',
+    offset = [0, -4],
+    children,
+}: MenuPopoverProps): JSX.Element | null => {
+    const [popperElement, setPopperElement] =
+        React.useState<HTMLDivElement | null>(null);
+
+    const {
+        styles: {popper},
+        attributes,
+        update,
+    } = usePopper(triggerRef.current, popperElement, {
```

Comment:
IMO we should be using floatingUI instead for 2 reasons:

1. popper is a discontinued package
2. floatingUI is a bit more comfortable to use

Reply:
this would go against the idea of only introducing new packages when absolutely necessary. Popperjs is widely used in mattermost webapp __

Reply:
I implemented floatingUI into another PR already #10529 ... so we can safely use it here as well. It is just a suggestion though and if we decide to pick that up we definitely need to replace all usages pf popper with floatingUI.

IMO the floatingUI package is a lot better in terms of usability, adjustability and style.

Reply:
Let's stick with popper here for now. We need to figure out where we're going with these since, even if floating-ui is better, we need to stick with a single dependency for some of these things